### PR TITLE
(almost) Stop using deprecated subscription NeoGo APIs

### LINKED ISSUE #2304: Node hangs after losing access to RPC nodes
## Expected Behavior
Node should either retry RPC connections forever or should shutdown properly (to be restarted by systemd) when it can't connect.

## Current Behavior
```
neofs-node[314962]: 2023-04-14T19:40:11.191Z        error        client/multi.go:113        could not establish connection to any RPC node
...
neofs-node[314962]: 2023-04-14T19:40:41.687Z        warn        subscriber/subscriber.go:136        remote notification channel has been closed
...
neofs-node[314962]: 2023-04-14T19:40:42.779Z        info        event/listener.go:228        stop event listener by context        {"reason": "context canceled"}
...
neofs-node[314962]: 2023-04-14T19:40:41.926Z        warn        neofs-node/main.go:137        internal application error        {"message": "morph connection has been lost"}
...
neofs-node[314962]: 2023-04-14T19:40:54.140Z        info        neofs-node/config.go:819        closing components of the storage engine...
...
neofs-node[314962]: 2023-04-14T19:42:28.638Z        info        neofs-node/config.go:827        all components of the storage engine closed successfully
neofs-node[314962]: 2023-04-14T19:42:28.641Z        info        neofs-node/grpc.go:110        stopping gRPC server...        {"name": "NeoFS Public API"}
neofs-node[314962]: 2023-04-14T19:42:29.074Z        info        neofs-node/grpc.go:89        stop listening gRPC endpoint        {"endpoint": "65.111.111.111:8080"}
neofs-node[314962]: 2023-04-14T19:42:29.074Z        info        neofs-node/grpc.go:126        gRPC server stopped successfully        {"name": "NeoFS Public API"}
neofs-node[314962]: 2023-04-14T19:42:29.084Z        info        neofs-node/grpc.go:110        stopping gRPC server...        {"name": "NeoFS Public API"}
neofs-node[314962]: 2023-04-14T19:42:29.118Z        info        neofs-node/grpc.go:126        gRPC server stopped successfully        {"name": "NeoFS Public API"}
neofs-node[314962]: 2023-04-14T19:42:29.118Z        info        neofs-node/grpc.go:89        stop listening gRPC endpoint        {"endpoint": "65.111.111.111:8082"}
```
And it hangs forever afterwards.

## Context
Happens on testnet, just happens. Both with 0.35.0 and 0.36.0.

## Regression
Seems like 0.34.0 behaved better.

## Your Environment
* Version used: 0.36.0
* Server setup and configuration: testnet storage node

### PR DESCRIPTION
**Pull request by roman-khimov**
Part of #2219, fixes #2304. Enough for 0.36.1 and we have a set of related tasks to do for 0.37.0.

### CODE REVIEW

File: cmd/neofs-node/config.go [hunk 1]
```diff
@@ -559,7 +559,7 @@ func initCfg(appCfg *config.Config) *cfg {
 	c.internals = internals{
 		ctx:          context.Background(),
 		appCfg:       appCfg,
-		internalErr:  make(chan error),
+		internalErr:  make(chan error, 10), // We only need one error, but we can have multiple senders.
```
Comment (cthulhu-rider) [inline cmd/neofs-node/config.go:562]: looks like a linear retraction of the problem to me, although I believe that in practice it is now quite working

if we really
> need one error

maybe we'll have channel with capacity = 1 and write errors w/o blocking?
```go
select {
  case internalErr <- err:
  default:
    // return or smth else
}
```

btw who closes the channel?

Comment (roman-khimov) [reply cmd/neofs-node/config.go:562]: https://github.com/nspcc-dev/neofs-node/blob/6f7127a9ea141b8cfd490c8cdc5284cf27dbff73/cmd/neofs-node/main.go#L133-L134
And that's a problem at the moment. The other approach is leaving some drain routine, but that won't exit as well.

File: pkg/morph/subscriber/subscriber.go [hunk 2]
```diff
@@ -199,16 +163,161 @@ func New(ctx context.Context, p *Params) (Subscriber, error) {
...
+	// into corresponding permanent channels.
 	go sub.routeNotifications(ctx)
 
 	return sub, nil
 }
 
+func (s *subscriber) routeNotifications(ctx context.Context) {
+	var (
+		// TODO: not needed after nspcc-dev/neo-go#2980.
+		cliCh             = s.client.NotificationChannel()
```
Comment (AnnaShaleva) [inline pkg/morph/subscriber/subscriber.go:184]: We may safely remove this old shared channel now.

File: pkg/morph/subscriber/subscriber.go [hunk 3]
```diff
@@ -199,16 +163,161 @@ func New(ctx context.Context, p *Params) (Subscriber, error) {
...
+				zap.Error(err),
+			)
+			resCh <- false
+			return
+		}
+	}
+
+	// notification events restoration
+	for contract := range s.subscribedEvents {
+		contract := contract // See https://github.com/nspcc-dev/neo-go/issues/2890
```
Comment (AnnaShaleva) [inline pkg/morph/subscriber/subscriber.go:295]: But why do we need it? We copy all the provided filters inside neo-go library, see https://github.com/nspcc-dev/neo-go/blob/7b109586ca75e50cdd56a76a68955a7506e9eda3/pkg/neorpc/filters.go#L82.

File: pkg/morph/subscriber/subscriber.go (hunk 2 above)
Comment (AnnaShaleva) [reply pkg/morph/subscriber/subscriber.go:184]: Then we need to add `no-lint` comment to the usage of deprecated WSClient's shared channel.

File: pkg/morph/subscriber/subscriber.go (hunk 3 above)
Comment (roman-khimov) [reply pkg/morph/subscriber/subscriber.go:295]: That's #2221, but we still have 0.100.1 used in mainnet (RPC nodes).

File: cmd/neofs-node/config.go (hunk 1 above)
Comment (cthulhu-rider) [reply cmd/neofs-node/config.go:562]: ah, reader closes the channel, always not so good

 idk, drop closing and non-blocking writing LGTM

File: pkg/morph/client/client.go [hunk 4]
```diff
@@ -496,26 +488,9 @@ func (c *Client) IsValidScript(script []byte, signers []transaction.Signer) (res
...
-	c.switchLock.Lock()
-	defer c.switchLock.Unlock()
-
-	close(c.notifications)
-	c.inactive = true
-
-	if c.cfg.inactiveModeCb != nil {
-		c.cfg.inactiveModeCb()
-	}
+	return c.client.Notifications
```
Comment (AnnaShaleva) [inline pkg/morph/client/client.go:493]: We need `nolint` comment here.

**Review by AnnaShaleva — COMMENTED** [review]
Otherwise LGTM.

**Review by cthulhu-rider — APPROVED** [review]

**Review by AnnaShaleva — APPROVED** [review]

**Pull request merged**